### PR TITLE
Desktop: Add Meaningful Error Messages for libdivecomputer Dump.

### PR DESCRIPTION
**Pull request by mikeller**
### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Add meaningful error messages when creating a libdivecomputer dump. In particular show if creating a dump is not supported on the dive computer that is used.

### Changes made:

1. add meaningful error messages when creating a libdivecomputer dump.

### Related issues:

https://github.com/subsurface/libdc/pull/51

### Additional information:

image

### CODE REVIEW

File: core/libdivecomputer.c [hunk 1]
```diff
@@ -1125,6 +1125,7 @@ static const char *do_device_import(device_data_t *data)
 	if (rc != DC_STATUS_SUCCESS)
 		return translate("gettextFromC", "Error registering the cancellation handler.");
 
+	const char *result = NULL;
```
Comment (bstoeger) [inline core/libdivecomputer.c]: What's the point of this more complex control flow? Because it is a C-File without RAII?

Comment (mikeller) [reply core/libdivecomputer.c]: I did this more 'out of habit' than anything else - in general I find that code without early returns throughout the functionality is easier to maintain once the functions become large and complex. But in this case it's really just an 'error out' pattern and could be changed back to early returns.

Comment (bstoeger) [reply core/libdivecomputer.c]: I thought the original was much easier to follow.

Yes, in C this technique may be needed to clean up resources, but here thankfully that is not the case.

If early returns become too complex,  it's probably time to split into multiple functions. In fact, I have seen people use local lambdas to emulate early return-like behavior.

Comment (mikeller) [reply core/libdivecomputer.c]: I'll change it back.

**Comment by bstoeger** [comment]
Thanks.

**Pull request merged**